app/services/doc_auth/mock/yml_loader_concern.rb

```diff
@@ -8,10 +8,10 @@ module YmlLoaderConcern
       def parse_yaml(uploaded_file)
         data = YAML.safe_load(uploaded_file, permitted_classes: [Date])
         if data.is_a?(Hash)
-          if (m = data.dig('document', 'dob').to_s.
-            match(%r{(?<month>\d{1,2})/(?<day>\d{1,2})/(?<year>\d{4})}))
-            data['document']['dob'] =
-              Date.new(m[:year].to_i, m[:month].to_i, m[:day].to_i)
+          ['dob', 'state_id_expiration'].each do |date_key|
+            if (date_s = data.dig('document', date_key))
+              data['document'][date_key] = DateParser.parse_legacy(date_s)
+            end
```

Comment:
Hah! I just got done writing a very similar code block here independently while working on LG-13517. I think our PRs are going to mesh nicely. Excellent.